### PR TITLE
refactor: watcher wip

### PR DESCRIPTION
**Pull request by nchanged**

### CODE REVIEW

File: playground/angular_playground/src/app/app.component.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { OtherService } from './services/OtherService';
   styleUrls: ['./app.component.scss'],
 })
 export class AppComponent {
-  title = 'hello world! YO';
+  title = 'hello!1!!';
```
Comment (FKobus) [inline playground/angular_playground/src/app/app.component.ts]: lol:P

File: src/output/distWriter.ts [hunk 2]
```diff
@@ -85,10 +85,11 @@ export function distWriter(props: IOuputParserProps): DistWriter {
       };
     },
     // the actual write function
-    write: async (target: string, contents: Buffer | string) => {
+    write: async (target: string, contents: Buffer | string): Promise<string> => {
       if (!path.isAbsolute(target)) target = path.join(self.outputDirectory, target);
       ensureDir(path.dirname(target));
-      writeFile(target, contents);
+      await writeFile(target, contents);
```
Comment (FKobus) [inline src/output/distWriter.ts]: add comment // @todo: try {} catch?

File: src/core/fusebox.ts [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
 
 export function fusebox(publicConfig: IPublicConfig) {
-  function execute(props: ICreateContextProps) {
+  async function execute(props: ICreateContextProps) {
     const ctx = createContext(props);
+
+    ctx.isWorking = true;
+    ctx.ict.sync('init', { ctx: ctx });
     preflightFusebox(ctx);
     switch (props.envType) {
```
Comment (FKobus) [inline src/core/fusebox.ts]: also add a comment here with `//todo try {} catch`

File: src/output/distWriter.ts (hunk 2 above)
Comment (nchanged) [reply src/output/distWriter.ts]: I think this one needs to be caught at a higher level...

Comment (FKobus) [reply src/output/distWriter.ts]: Okay, let's figure out how we want to handle these later

File: src/core/fusebox.ts (hunk 3 above)
Comment (nchanged) [reply src/core/fusebox.ts]: done, i"ve wrapped it into a try catch block

**Comment by FKobus** [comment]
:+1:

**Pull request merged**